src/main/java/com/serotonin/bacnet4j/obj/ScheduleObject.java

```diff
@@ -340,7 +341,7 @@ private void updatePresentValue(final DateTime now) {
                 }
 
                 // Determine the new present value.
-                if (currentTv == null)
+                if (currentTv == null || currentTv.getValue().getClass().equals(Null.class))
```

Comment:
So where does the schedule check the next "layer" if the current schedule's value is null?  My understanding is that the spec states you need to loop over the schedules in priority order somewhere and select the highest priority value for that time slot that isn't Null.

Reply:
private void updatePresentValue(final DateTime now) should check the following sequentially
1. parse the Exception Schedule:
      1.1 - Get the Time Value Pair (TV) of the current day along with their priority
       1.2 - There are chanceupdatePresentValue()s on the same time there could be two values say V1 and V2 
               configiured in diff priorities say 1 and  2.  Please note these prio are part of Exception Schedule. So in this 
              case, the currentTV = V1 as the prio is 1. 
        1.3 - Though the current code checks for special Event, but does it check if there are multiple entries on the 
                 given day. Say for example in the above case V1 is 10:00AM to 2PM, and V2 is 8:00AM to 8:00PM. 
                 In this example at 8:00AM currentTV = (8:00AM, V2) & the nextCheck = 10:00AM. 
                 At 10:00AM, currentTV = (10:00, V1) and nextCheck = 2 PM. 
                At 2:00PM, the currentTV = (2: PM, V2), the nextCheck = 8PM. At the 8PM, specialEvent=Null resulting from 
                findExceptionSchedule(now) 
But I doubt the current code work similar to the above situation. 
2. parse DailyScheule:
        2.1 - dailySchedule.getDayScheule() - whether it returns the TV pair in ascending / descending order needs to be 
                checked. 
3. ScheduleDefault will be in active when the result of findExceptionSchedule(now) is NULL. 

I think most of the points that needs to be addressed are mentioned above. 
 
Regards,
Kishore

Reply:
In validateProperty() function, exception is thrown when the ScheduleDefault.class is not matching with the class of currentTV. 
1. But either the Schedule Default Value can have NULL or TV pair can have NULL. 
2. In site, say Schedule,1 is configured for AnalogValue. Later they may configure it BinaryPV as well by changing the parameters one by one like WklySchedule, ExcepSchule, ScheduleDefault etc. In the current situation they will not be in a position to do it. Once the schedule is configured its locked for that data type only. 
3. Hence instead of throwing exception, reliability property can be set to Config-Error untill all the datatypes are matching. 

Kishore

Reply:
@kishorevenki I am going to create another PR to handle NULL value for Schedule Default.